src/Symfony/Component/Workflow/EventListener/GuardListener.php

```diff
@@ -28,14 +29,23 @@ class GuardListener
     private $authenticationChecker;
     private $trustResolver;
     private $roleHierarchy;
+    private $validator;
 
-    public function __construct($configuration, ExpressionLanguage $expressionLanguage, TokenStorageInterface $tokenStorage, AuthorizationCheckerInterface $authenticationChecker, AuthenticationTrustResolverInterface $trustResolver, RoleHierarchyInterface $roleHierarchy = null)
-    {
+    public function __construct(
+        $configuration,
+        ExpressionLanguage $expressionLanguage,
+        TokenStorageInterface $tokenStorage,
+        AuthorizationCheckerInterface $authenticationChecker,
+        AuthenticationTrustResolverInterface $trustResolver,
+        ValidatorInterface $validator,
+        RoleHierarchyInterface $roleHierarchy = null
```

Comment:
1. should be on one line
2. could you put the validator at the last position? 
3. could you make the validator optional to preserve the BC + a runtime exception is the validator is not defined but the user want to use it.